x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/version/EsqlVersion.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.version;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.VersionId;
+
+public enum EsqlVersion implements VersionId<EsqlVersion> {
+    // Breaking changes go here until the next version is released.
+    NIGHTLY(Integer.MAX_VALUE, Integer.MAX_VALUE, "\uD83D\uDE34"),
+    PARTY_POPPER(2024, 4, "\uD83C\uDF89");
+
+    EsqlVersion(int year, int month, String emoji) {
```

Comment:
If we think we might need to serialize this, we should explicitly pass in an ordinal value for serialization (that way it's proof against re-ordering the constants or deleting one).  I don't know that we'll have to serialize these, but in case you aren't familiar with that pattern, I wanted to call it out.